collector/lib/DriverCandidates.h

```diff
@@ -0,0 +1,70 @@
+/** collector
+
+A full notice with attributions is provided along with this source code.
+
+This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License version 2 as published by the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+* In addition, as a special exception, the copyright holders give
+* permission to link the code of portions of this program with the
+* OpenSSL library under certain conditions as described in each
+* individual source file, and distribute linked combinations
+* including the two.
+* You must obey the GNU General Public License in all respects
+* for all of the code used other than OpenSSL.  If you modify
+* file(s) with this exception, you may extend this exception to your
+* version of the file(s), but you are not obligated to do so.  If you
+* do not wish to do so, delete this exception statement from your
+* version.
+*/
+
+#ifndef _DRIVER_CANDIDATES_H_
+#define _DRIVER_CANDIDATES_H_
+
+#include <string>
+#include <vector>
+
+namespace collector {
+
+class DriverCandidate {
+ public:
+  DriverCandidate(const std::string& name, bool useEbpf, const std::string& shortName = "", const std::string& path = "/kernel-modules", bool downloadable = true) : name_(name), shortName_(shortName), path_(path), downloadable_(downloadable) {
```

Comment:
What about passing a `collectionMethod` enum value here instead of `bool`? Would probably make it easier to extend in the future.

Reply:
I thought about this and decided we could do it in a follow up, but maybe I just make the `collectionMethod` enum public and start using it here. I just need to figure out if it makes sense to leave it in `DriverCandidates.h` or if I should move it somewhere else.

Reply:
Good question, but I agree with doing it in the follow up.